apps/grader-host/src/index.ts

```diff
@@ -572,17 +643,25 @@ async function uploadArchive(context: Context, tempDir: string) {
         Body: fs.createReadStream(archiveFile.path),
       },
     }).done();
-
-    // Upload all logs to S3.
-    await new Upload({
-      client: s3,
-      params: {
-        Bucket: s3Bucket,
-        Key: `${s3RootKey}/output.log`,
-        Body: logger.getBuffer(),
-      },
-    }).done();
   } finally {
     await archiveFile.cleanup();
   }
 }
+
+async function uploadLogs(context: Context) {
```

Comment:
I split this out so that we'd always attempt to upload logs even if other things failed.